src/lib.rs

```diff
@@ -310,13 +312,54 @@ pub fn select_coin_srd(
     })
 }
 
-/// The Global Coinselection API that performs all the algorithms and proudeces result with least [WasteMetric].
+/// The Global Coinselection API that performs all the algorithms and produces result with least [WasteMetric].
 /// At least one selection solution should be found.
+type SelectionFn = fn(&[OutputGroup], CoinSelectionOpt) -> Result<SelectionOutput, SelectionError>;
 pub fn select_coin(
     inputs: &[OutputGroup],
     options: CoinSelectionOpt,
 ) -> Result<SelectionOutput, SelectionError> {
-    unimplemented!()
+    let best_result: Arc<Mutex<Result<SelectionOutput, SelectionError>>> =
+        Arc::new(Mutex::new(Err(SelectionError::NoSolutionFound)));
+
+    // Vector of selection functions
+    let selection_fns: Vec<SelectionFn> =
+        vec![select_coin_fifo, select_coin_lowestlarger, select_coin_srd];
+
+    let handles: Vec<_> = selection_fns
+        .into_iter()
+        .map(|selection_fn| {
+            let best_result_clone = Arc::clone(&best_result);
+            let inputs_clone = inputs.to_vec();
+            let options_clone = options;
+            thread::spawn(move || {
+                let result = selection_fn(&inputs_clone, options_clone);
+                if let Ok(output) = result {
+                    let mut best_result = best_result_clone.lock().unwrap();
+                    match &*best_result {
+                        Ok(best_output) => {
+                            if output.waste.0 < best_output.waste.0 {
+                                *best_result = Ok(output);
+                            }
+                        }
+                        Err(SelectionError::InsufficientFunds) => {
+                            *best_result = Err(SelectionError::InsufficientFunds)
```

Comment:
This doesn't make sense because you are checking whether best_result is Err(Insuff) and then updating it with the same value.